pkg/controller/machine.go

```diff
@@ -569,6 +569,9 @@ func (c *controller) machineDelete(machine *v1alpha1.Machine, driver driver.Driv
 		// If machine was created on the cloud provider
 		machineID, _ := driver.GetExisting()
 
+		// update node with the machine's state prior to termination
+		c.AnnotateTerminatingMachineNode(machine)
```

Comment:
> I guess, should be fine to ignore the error here, but then it becomes the best-effort feature, and not guruanteed. I would prefer to understand the consequences on both side.

I was concerned about adding something to the critical path of node deprovisioning. Since you approve, I will return an error here, to ensure that this is retried and the annotations is added (if the node exists). 

> Also, I'd be curious to know your plans of making use of this feature in general.

We run some single-replica deployments in our cluster. The associated pods must not be disrupted during cluster node upgrades, at least until some timeout is reached. The MCM drain performs voluntary evictions, so we use PDBs with `maxUnavailable: 0` to enforce this behavior. 

However, the same draining behavior would apply when replacing unhealthy nodes. We want to identify this situation, and automate the removal of PDBs from unhealthy nodes that are being deprovisioned, to ensure that they are removed promptly.

Reply:
Sure, sounds good.